Snapshot release failing with no `IREE_ELF_CLEAR_CACHE` on gcc

```
iree/hal/local/elf/CMakeFiles/iree_hal_local_elf_platform.dir/platform/linux.c.o -c /project/main_checkout/iree/hal/local/elf/platform/linux.c
2021-04-26T11:34:03.1538953Z /project/main_checkout/iree/hal/local/elf/platform/linux.c:155:2: error: #error "no instruction cache clear implementation"
2021-04-26T11:34:03.1540158Z   155 | #error "no instruction cache clear implementation"
2021-04-26T11:34:03.1540883Z       |  ^~~~~
2021-04-26T11:34:03.1542565Z /project/main_checkout/iree/hal/local/elf/platform/linux.c: In function ‘iree_memory_view_flush_icache’:
2021-04-26T11:34:03.1545027Z /project/main_checkout/iree/hal/local/elf/platform/linux.c:160:3: warning: implicit declaration of function ‘IREE_ELF_CLEAR_CACHE’ [-Wimplicit-function-declaration]
2021-04-26T11:34:03.1546548Z   160 |   IREE_ELF_CLEAR_CACHE(base_address, base_address + length);
2021-04-26T11:34:03.1567057Z       |   ^~~~~~~~~~~~~~~~~~~~
```

--- a/iree/hal/local/elf/platform/generic.c
+++ b/iree/hal/local/elf/platform/generic.c
@@ -94,8 +94,10 @@ iree_status_t iree_memory_view_protect_ranges(void* base_address,
 // IREE_ELF_CLEAR_CACHE can be defined externally to override this default
 // behavior.
 #if !defined(IREE_ELF_CLEAR_CACHE)
+// Explicitly enable for GCC, which has had this since 4.x but does not
+// seem to advertise it via __has_builtin.
 #if defined __has_builtin
-#if __has_builtin(__builtin___clear_cache)
+#if __has_builtin(__builtin___clear_cache) || defined(__GNUC__)
 #define IREE_ELF_CLEAR_CACHE(start, end) __builtin___clear_cache(start, end)
 #endif  // __builtin___clear_cache
 #endif  // __has_builtin

--- a/iree/hal/local/elf/platform/linux.c
+++ b/iree/hal/local/elf/platform/linux.c
@@ -144,8 +144,10 @@ iree_status_t iree_memory_view_protect_ranges(void* base_address,
 // IREE_ELF_CLEAR_CACHE can be defined externally to override this default
 // behavior.
 #if !defined(IREE_ELF_CLEAR_CACHE)
+// Explicitly enable for GCC, which has had this since 4.x but does not
+// seem to advertise it via __has_builtin.
 #if defined __has_builtin
-#if __has_builtin(__builtin___clear_cache)
+#if __has_builtin(__builtin___clear_cache) || defined(__GNUC__)
 #define IREE_ELF_CLEAR_CACHE(start, end) __builtin___clear_cache(start, end)
 #endif  // __builtin___clear_cache
 #endif  // __has_builtin

